src/renderer/store/takeGroups/reducer.ts

```diff
@@ -25,6 +29,12 @@ const takeGroupsReducer: Reducer<TakeGroup[], Action<any>> = (
     );
   }
 
+  if (action.type === UNDO_SELECT_TAKE) {
+    const { takeGroup } = action.payload as UndoSelectTakePayload;
+
+    return takeGroups.map((tg) => (tg.id === takeGroup.id ? takeGroup : tg));
```

Comment:
Rather than overriding entire take group, wouldn't it make more sense just to store the originally active take ID (or none if there was no selected take in the group) and change the active one to the one that was active before?

Reply:
We need to undo both activeTakeIndex (which denotes the selected take) and takeSelected (boolean denoting if a take has been selected yet) and we need the id to find it, which are all the values contained in TakeGroup anyway.

Reply:
Yep okay, this is fine then. If we weren't in the second last week though I'd say account for possible extensions of the TakeGroup because if the type ever changes this code will break